sdk/python/feast/client.py

```diff
@@ -1113,8 +1117,86 @@ def start_stream_to_online_ingestion(
             )
             response = self._job_service.StartStreamToOnlineIngestionJob(request)
             return RemoteStreamIngestionJob(
-                self._job_service, self._extra_grpc_params, response.id,
+                self._job_service, self._extra_grpc_params, response.id
+            )
+
+    def _get_expected_job_hash_to_table_refs(
+        self, all_projects: bool
+    ) -> Dict[str, Tuple[str, str]]:
+        """
+        Checks all feature tables for the requires project(s) and determines all required stream
+        ingestion jobs from them. Outputs a map of the expected job_hash to a tuple of (project, table_name).
+
+        Args:
+            all_projects (bool): If true, runs the check for all project.
+                Otherwise only checks the current project.
+
+        Returns:
+            Dict[str, Tuple[str, str]]: Map of job_hash -> (project, table_name) for expected stream ingestion jobs
+        """
+        job_hash_to_table_refs = {}
+
+        projects = self.list_projects() if all_projects else [self.project]
+        for project in projects:
+            feature_tables = self.list_feature_tables(project)
+            for feature_table in feature_tables:
+                if feature_table.stream_source is not None:
+                    params = get_stream_to_online_ingestion_params(
+                        self, project, feature_table, []
+                    )
+                    job_hash = params.get_job_hash()
+                    job_hash_to_table_refs[job_hash] = (project, feature_table.name)
+
+        return job_hash_to_table_refs
+
+    def ensure_stream_ingestion_jobs(self, all_projects: bool = False):
+        """Ensures all required stream ingestion jobs are running and cleans up the unnecessary jobs.
+
+        More concretely, it will determine
+        - which stream ingestion jobs are running
+        - which stream ingestion jobs should be running
+        And it'll do 2 kinds of operations
+        - Cancel all running jobs that should not be running
+        - Start all non-existent jobs that should be running
+
+        Args:
+            all_projects (bool, optional): If true, runs the check for all project.
```

Comment:
`run the check` is a bit vague, maybe `ensures stream ingestion jobs are running for all projects`?